python/triton/runtime/jit.py

```diff
@@ -359,9 +359,35 @@ def _type_of(key, is_const=False):
             return key
 
         dtype_str = str(key).split(".")[-1]
-        dtype_str = type_canonicalisation_dict[dtype_str]
-        const_str = "*k" if is_const else "*"
-        return const_str + dtype_str
+        tys = {
+            "bool": "i1",
+            "float8e4nv": "fp8e4nv",
+            "float8e4b8": "fp8e4b8",
+            "float8e5": "fp8e5",
+            "float8e5b16": "fp8e5b16",
+            "float8e4b15": "fp8e4b15",
+            "float8_e4m3fn": "fp8e4nv",
+            "float8_e4m3fnuz": "fp8e4b8",
+            "float8_e5m2": "fp8e5",
+            "float8_e5m2fnuz": "fp8e5b16",
+            "float16": "fp16",
+            "bfloat16": "bf16",
+            "float32": "fp32",
+            "float64": "fp64",
+            "int8": "i8",
+            "int16": "i16",
+            "int32": "i32",
+            "int64": "i64",
+            "uint8": "u8",
+            "uint16": "u16",
+            "uint32": "u32",
+            "uint64": "u64",
```

Comment:
is this a duplicate of `type_canonicalisation_dict`? It would be better to have only one source of truth

Reply:
Seems that the duplication is already removed in the latest upstream. We will rebase to the latest.

Reply:
Rebased.

Reply:
> Can we actually directory use `type_canonicalisation_list` here? It's the same to me.

The duplication was already removed by rebasing to latest upstream.

Reply:
Oops, I missed checking this one. I pushed a change to remove the duplication.

Reply:
Hmm, latest main had this duplicate removed and I thought I had also adopted that. Somehow it turned out that it didn't... Thanks @antiagainst for cleaning this up!

Reply:
no worries!